tests/test_mypy.py

```diff
@@ -0,0 +1,164 @@
+    bc = Breadcrumb(url=response.url)
+    bc.url = response.url
+    bc.url = str(response.url)
+    bc.url = 123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+
+
+@pytest.mark.mypy_testing
+def test_assignment_link():
+    link = Link(url=response.url)
+    link.url = response.url
+    link.url = str(response.url)
+    link.url = 123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+
+
+@pytest.mark.mypy_testing
+def test_assignment_product():
+    product = Product(url=response.url)
+    product.url = response.url
+    product.url = str(response.url)
+    product.url = 123  # E: Incompatible types in assignment (expression has type "int", variable has type "str")
+    product.canonicalUrl = response.url
+    product.canonicalUrl = str(response.url)
+    product.canonicalUrl = (
+        123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+    )
+
+
+@pytest.mark.mypy_testing
+def test_assignment_product_list():
+    product_list = ProductList(url=response.url)
+    product_list.url = response.url
+    product_list.url = str(response.url)
+    product_list.url = 123  # E: Incompatible types in assignment (expression has type "int", variable has type "str")
+    product_list.canonicalUrl = response.url
+    product_list.canonicalUrl = str(response.url)
+    product_list.canonicalUrl = (
+        123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+    )
+
+
+@pytest.mark.mypy_testing
+def test_assignment_product_from_list():
+    product_from_list = ProductFromList(url=response.url)
+    product_from_list.url = response.url
+    product_from_list.url = str(response.url)
+    product_from_list.url = (
+        123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+    )
+
+
+@pytest.mark.mypy_testing
+def test_assignment_product_variant():
+    product_variant = ProductVariant(url=response.url)
+    product_variant.url = response.url
+    product_variant.url = str(response.url)
+    product_variant.url = (
+        123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+    )
+    product_variant.canonicalUrl = response.url
+    product_variant.canonicalUrl = str(response.url)
+    product_variant.canonicalUrl = (
+        123  # E: Incompatible types in assignment (expression has type "int", variable has type "Optional[str]")
+    )
+
+
+# NOTE: The tests below are marked to fail since the following error is
+# encountered in mypy:
+#
+#   "error: Skipping analyzing "web_poet": module is installed, but missing
+#   library stubs or py.typed marker"
+#
+# This means that mypy renders all contents (i.e. URL classes) from web-poet to
+# be of type Any. To fix this, web-poet must either be PEP 561 complient with
```

Comment:
If web-poet code is annotated, we should add py.typed marker.

Reply:
Actually, we should do it for zyte-common-items as well :)

Reply:
Created an issue in `web-poet` here: https://github.com/scrapinghub/web-poet/issues/65.

Should we also handle https://peps.python.org/pep-0561/ in this PR for `zyte-common-items`?

Reply:
Thanks @BurnzZ! No need to handle it in this PR; I've created https://github.com/zytedata/zyte-common-items/pull/12 and https://github.com/scrapinghub/web-poet/pull/66.

Reply:
We can't release web-poet yet though, because of unfinished fields feature.

Reply:
Thanks @kmike ! 

> We can't release web-poet yet though, because of unfinished fields feature.

That's okay, we can temporarily point to the commits for this PR to check that the tests are working alright.